pkg/api/unversioned/group_version.go

```diff
@@ -198,3 +198,46 @@ func (gv *GroupVersion) UnmarshalJSON(value []byte) error {
 func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
+
+// ToAPIVersionAndKind is a convenience method for satisfying runtime.Object on types that
+// do not use TypeMeta.
+func (gvk *GroupVersionKind) ToAPIVersionAndKind() (string, string) {
+	if gvk == nil {
+		return "", ""
+	}
+	return gvk.GroupVersion().String(), gvk.Kind
+}
+
+// FromAPIVersionAndKind returns a GVK representing the provided fields for types that
+// do not use TypeMeta.
+func FromAPIVersionAndKind(apiVersion, kind string) *GroupVersionKind {
+	if gv, err := ParseGroupVersion(apiVersion); err == nil {
+		return &GroupVersionKind{Group: gv.Group, Version: gv.Version, Kind: kind}
+	}
+	return &GroupVersionKind{Kind: kind}
+}
+
+// All objects that are serialized from a Scheme encode their type information. This interface is used
+// by serialization to set type information from the Scheme onto the serialized version of an object.
+// For objects that cannot be serialized or have unique requirements, this interface may be a no-op.
+// TODO: this belongs in pkg/runtime, move unversioned.GVK into runtime.
```

Comment:
@caesarxuchao This comment relates to https://github.com/kubernetes/kubernetes/issues/17216#issuecomment-163787857.

@smarterclayton See the linked comment.